float32 theta_rad
float32 val
float32 time_sec